virtual_rainforest/models/animals/animal_model.py

```diff
@@ -55,6 +55,12 @@ def __init__(
         **kwargs: Any,
     ):
         super().__init__(data, update_interval, start_time, **kwargs)
+        from virtual_rainforest.models.animals.dummy_animal_module import AnimalCohort
+
+        self.cohort_positions: dict[int, list[AnimalCohort]] = {
+            k: [] for k in self.data.grid.cell_id
+        }
```

Comment:
Let me know if I'm understanding you correctly, you're suggesting replacing the dictionary with an unstructured list of all of the animal cohorts in the model?

The dictionary structure isn't really about dispersal, it's about how trophic interactions will work. Every tick there will be many calls to a search to figure out what other animal cohorts are possible food sources or other interactions. I wanted to keep that spatial info organized to make that process easier. 

Does that make sense?

Reply:
I've committed a basic version of what I think you're talking about. It still needs a couple of tests and, if we agree about the handling of births/deaths for those handler functions within `AnimalModel`. Let me know if this is, in fact, what you are talking about and if you have any thoughts about the trophic issue.

Reply:
It might be worth having a group chat about how to structure this. One option that might be worth exploring is to avoid having an additional layer in the class structure:

* **AnimalModel**: 
  * handles the setup
  * creates a list of AnimalCommunity objects, one per grid cell.
  * provides any additional setup for connectivity between grid cells if needed.
* **AnimalCommunity**:
  * is the main workhorse containing all of the behavioural functionality
  * has a single list of AnimalCohorts, which are all of the cohorts currently present in the grid cell
  * has an `immigrate` method, which allows an AnimalCohort instance to be popped out of one instance and added to the cohort list for another AnimalCommunity
* **AnimalCohort**:
   * maintains the functional type and status of the individual cohorts.
 
This explicitly separates the within cell ecology happening to a set of cohorts, and the cohorts within a cell are always up to date, as the cohort list updates in real time with immigration.

Reply:
I think most of what is in `AnimalModel` now would move into `AnimalCommunity`?

Reply:
Interesting idea! I think that will be a cleaner way to organize things. I would rather have those function calls in something like an `AnimalCommunity` class than cluttering up `AnimalModel`. I'll throw something together and we can see what we think.

Reply:
It might make sense if each `AnimalCommunity` had a dictionary of cohorts keyed by functional type, rather than just a flat list? So:

```python
my_animal_community.cohorts = {
    FuncGroup1: [fg1cohort1, fg1cohort2], 
    FuncGroup2: [fg2cohort1, fg2cohort2, fg2cohort3],
}
```

Reply:
After a bit of battling with naming, I think I've got things smoothed out! Take a look and let me know if this is what you had in mind with an intermediate 'AnimalCommunity' class. It seems to have worked well.